### PR TITLE
Update Pod overview

### PR DESCRIPTION
**Pull request by tengqm**
This PR adds some clarifications about Pod update and Pod replacement, which can be pretty confusing to new users.

### CODE REVIEW

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 1]
```diff
@@ -191,6 +191,30 @@ details are abstracted away. That abstraction and separation of concerns simplif
 system semantics, and makes it feasible to extend the cluster's behavior without
 changing existing code.
 
+## Pod update and replacement
+
+As mentioned in the previous section, when the Pod template for a workload
+resource is changed, the controller will create new Pods based on the updated
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:197]:
```suggestion
resource is changed, the controller creates new Pods based on the updated
```

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 2]
```diff
@@ -191,6 +191,30 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+template instead of updating or patching the existing Pods.
+
+Kubernetes doesn't prevent you from managing Pods directly. It is possible to
+update a Pod in place. However, the Pod update operation has some limitations:
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:201]:
```suggestion
update some fields of an existing Pod, in place. However, the Pod update operation has some limitations:
```
?

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 3]
```diff
@@ -191,6 +191,30 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+
+- Most of the metadata about a Pod is immutable. For example, you cannot
+  change the `namespace`, `name`, `uid`, `creationTimestamp` fields, you can
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:204]:
```suggestion
  change the `namespace`, `name`, `uid`, `creationTimestamp` fields; you can
```

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 4]
```diff
@@ -191,6 +191,30 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+  only increment the `generation` field if you want to change it.
+- If the `metadata.deletionTimestamp` is set, no new entry can be added to the
+  `metadata.finalizers` list.
+- Pod updates may not change fields other than `spec.containers[*].image`,
+  `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or
+  `spec.tolerations`. Only adding new item into `spec.tolerations` is
+  permitted.
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:211]:
```suggestion
   `spec.tolerations`. For `spec.tolerations`, you can only add new entries.
```

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 5]
```diff
@@ -191,6 +191,30 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+- When updating the `spec.activeDeadlineSeconds` field, two types of updates
+  are allowed:
+
+  1. update the field from nil to a positive value;
+  1. update the field from a positive value to a smaller, non-negative value.
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:216]:
```suggestion
  1. setting the field (updating the field from nil to a positive number);
  1. updating the field from a positive number to a smaller, non-negative number.
```
?

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 6]
```diff
@@ -191,6 +191,30 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+template instead of updating or patching the existing Pods.
+
+Kubernetes doesn't prevent you from managing Pods directly. It is possible to
+update a Pod in place. However, the Pod update operation has some limitations:
+
+- Most of the metadata about a Pod is immutable. For example, you cannot
+  change the `namespace`, `name`, `uid`, `creationTimestamp` fields, you can
+  only increment the `generation` field if you want to change it.
+- If the `metadata.deletionTimestamp` is set, no new entry can be added to the
+  `metadata.finalizers` list.
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:207]: Maybe mention that this is not specific to Pods (but! I can't think of some concise wording).

**Review by sftim — COMMENTED** [review]
/lgtm

Also some feedback, feel free to accept only the comments that you like the look of.

File: content/en/docs/concepts/workloads/pods/_index.md (hunk 2 above)
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/_index.md:201]: nit: Does it make sense to replace "in place" with: 
It is possible to update a running Pod.

nit: What exactly is the Pod "update operation"?
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#patch-pod-v1-core

File: content/en/docs/concepts/workloads/pods/_index.md (hunk 3 above)
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/_index.md:204]: nit?: For example, you cannot change the `namespace`, `name`, `uid`, or `creationTimestamp` fields.
nit: The `generation` field is unique. This field only accepts updates that increment the field's current value?

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 7]
```diff
@@ -191,6 +191,30 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+- Pod updates may not change fields other than `spec.containers[*].image`,
+  `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or
+  `spec.tolerations`. Only adding new item into `spec.tolerations` is
+  permitted.
+- When updating the `spec.activeDeadlineSeconds` field, two types of updates
+  are allowed:
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/_index.md:213]: @sftim suggestions look good. Does the `nil` setting imply that the field is unset or possibly set to 0, ""?
You can change the `spec.activeDeadlineSeconds` field by:
- modifying the field from `nil` to a positive number, or
- modifying the field from a positive number ...

File: content/en/docs/concepts/workloads/pods/_index.md (hunk 6 above)
Comment (tengqm) [reply content/en/docs/concepts/workloads/pods/_index.md:207]: Maybe this is already done in the GC concept/task.

File: content/en/docs/concepts/workloads/pods/_index.md (hunk 2 above)
Comment (tengqm) [reply content/en/docs/concepts/workloads/pods/_index.md:201]: update include 'patch' and 'replace', revised.

File: content/en/docs/concepts/workloads/pods/_index.md (hunk 7 above)
Comment (tengqm) [reply content/en/docs/concepts/workloads/pods/_index.md:213]: nil is unset.

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 8]
```diff
@@ -191,6 +191,35 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+- If the `metadata.deletionTimestamp` is set, no new entry can be added to the
+  `metadata.finalizers` list.
+- Pod updates may not change fields other than `spec.containers[*].image`,
+  `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or
+  `spec.tolerations`. For `spec.tolerations`, you can only add new entries.
+- When updating the `spec.activeDeadlineSeconds` field, two types of updates
+  are allowed:
+
+  1. setting the field (updating the field from nil to a positive number);
+  1. updating the field from a positive nubmer to a smaller, non-negative
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/_index.md:220]: nit: s/nubmer/number/

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 9]
```diff
@@ -191,6 +191,35 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
 changing existing code.
 
+## Pod update and replacement
+
+As mentioned in the previous section, when the Pod template for a workload
+resource is changed, the controller creates new Pods based on the updated
+template instead of updating or patching the existing Pods.
+
+Kubernetes doesn't prevent you from managing Pods directly. It is possible to
+update some fields of an running Pod, in place. However, the Pod update
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/_index.md:201]: nit: a running Pod ...

File: content/en/docs/concepts/workloads/pods/_index.md (hunk 7 above)
Comment (sftim) [reply content/en/docs/concepts/workloads/pods/_index.md:217]: nil is a bit surprising. On the wire it's JSON and the field would be _I think_ `null` or just not present. If you used `kubectl edit replicaset/foo` (say) then you'd get YAML and again there's either `null` or you omit the field.

In the source code for the API server, it's `nil`. But: Kubernetes tries to be implementation neutral; if someone wants to write a 100% conformant Kubernetes API server in Python, more strength to them.

**Review by sftim — COMMENTED** [review]
/lgtm

One quibble about exposing `nil` as an implementation detail; we usually avoid revealing details about the Go implementation if they're not part of the API.

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 10]
```diff
@@ -191,6 +191,35 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+  field's current value.
+- If the `metadata.deletionTimestamp` is set, no new entry can be added to the
+  `metadata.finalizers` list.
+- Pod updates may not change fields other than `spec.containers[*].image`,
+  `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or
+  `spec.tolerations`. For `spec.tolerations`, you can only add new entries.
+- When updating the `spec.activeDeadlineSeconds` field, two types of updates
+  are allowed:
+
+  1. setting the field (updating the field from nil to a positive number);
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/_index.md:219]: nit: Yes, I think this case could be described as:
`setting the field (or updating the unassigned field) to a positive number;`

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 11]
```diff
@@ -191,6 +191,35 @@ details are abstracted away. That abstraction and separation of concerns simplif
...
+  field's current value.
+- If the `metadata.deletionTimestamp` is set, no new entry can be added to the
+  `metadata.finalizers` list.
+- Pod updates may not change fields other than `spec.containers[*].image`,
+  `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or
+  `spec.tolerations`. For `spec.tolerations`, you can only add new entries.
+- When updating the `spec.activeDeadlineSeconds` field, two types of updates
+  are allowed:
+
+  1. setting the unssigned field to a positive number;
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/_index.md:219]: nit: s/unssigned/unassigned/

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-25756--kubernetes-io-master-staging.netlify.app/docs/concepts/workloads/pods/#pod-update-and-replacement

/lgtm

**Comment by kbhawkey** [comment]
Question: Does this new section belong in a "Pod configuration" page or the Pod overview page?

**Comment by sftim** [comment]
Happy to see these improvement land
/approve

**Pull request merged**